packages/fiori/src/DynamicPageTitle.ts

```diff
@@ -0,0 +1,287 @@
+import { getI18nBundle } from "@ui5/webcomponents-base/dist/i18nBundle.js";
+import type I18nBundle from "@ui5/webcomponents-base/dist/i18nBundle.js";
+import ResizeHandler from "@ui5/webcomponents-base/dist/delegate/ResizeHandler.js";
+import type { ResizeObserverCallback } from "@ui5/webcomponents-base/dist/delegate/ResizeHandler.js";
+import { isEnter, isSpace } from "@ui5/webcomponents-base/dist/Keys.js";
+import type Toolbar from "@ui5/webcomponents/dist/Toolbar.js";
+import type { ToolbarMinWidthChangeEventDetail } from "@ui5/webcomponents/dist/Toolbar.js";
+import ToolbarItemOverflowBehavior from "@ui5/webcomponents/dist/types/ToolbarItemOverflowBehavior.js";
+import { isDesktop } from "@ui5/webcomponents-base/dist/Device.js";
+
+// Template
+import DynamicPageTitleTemplate from "./generated/templates/DynamicPageTitleTemplate.lit.js";
+
+// Styles
+import DynamicPageTitleCss from "./generated/themes/DynamicPageTitle.css.js";
+
+// Texts
+import {
+	DYNAMIC_PAGE_ARIA_DESCR_TOGGLE_HEADER,
+} from "./generated/i18n/i18n-defaults.js";
+
+/**
+ * @class
+ *
+ * ### Overview
+ *
+ * Title of the `DynamicPage`.
+ *
+ * The `DynamicPageTitle` component is part of the `DynamicPage`
+ * family and is used to serve as title of the `DynamicPage`.
+ *
+ * ### Usage
+ *
+ * The `DynamicPageTitle` can hold any component and displays the most important
+ * information regarding the object that will always remain visible while scrolling.
+ *
+ * **Note:** The `actions` slot accepts any UI5 web component, but it's
+ * recommended to use `ui5-toolbar`.
+ *
+ * The user can switch between the expanded/collapsed states of the
+ * `DynamicPage` by clicking on the `DynamicPageTitle`
+ * or by using the expand/collapse visual indicators, positioned at the bottom of the
+ * `DynamicPageTitle` and the `DynamicPageHeader` inside `ui5-dynamic-page-header-actions`.
+ *
+ * ### Responsive Behavior
+ *
+ * The responsive behavior of the `DynamicPageTitle` depends on the behavior of the
+ * content that is displayed.
+ * @constructor
+ * @extends UI5Element
+ * @public
+ * @since 2.0.0
+ */
+@customElement({
+	tag: "ui5-dynamic-page-title",
+	fastNavigation: true,
+	renderer: litRender,
+	styles: DynamicPageTitleCss,
+	template: DynamicPageTitleTemplate,
+})
+
+/**
+ * Event is fired when the title is toggled.
+ * @private
+ */
+@event("_toggle-title")
+
+class DynamicPageTitle extends UI5Element {
+	/**
+	 * Defines if the title is snapped.
+	 *
+	 * @protected
+	 * @default false
+	 */
+	@property({ type: Boolean })
+	snapped!: boolean;
+
+	/**
+	 * Defines if the mobileNavigationActions are shown.
+	 *
+	 * @private
+	 */
+	@property({ type: Boolean })
+	mobileNavigationActions!: boolean;
+
+	/**
+	 * Indicates if the elements is on focus
+	 * @private
+	 */
+	@property({ type: Boolean })
+	focused!: boolean;
+
+	/**
+	 * Defines the content of the Heading of the Dynamic Page.
+	 *
+	 * @public
```

Comment:
How we use heading, expendedHeading, snappedHeading, isn't only 2 states expended and snapped and two slots enough